src/awkward/_v2/_connect/rdataframe/from_rdataframe.py

```diff
@@ -43,140 +48,214 @@
 assert done is True
 
 
-def from_rdataframe(data_frame, column):
-    def _wrap_as_record_array(array):
-        layout = array.layout if isinstance(array, ak._v2.highlevel.Array) else array
-        return ak._v2._util.wrap(
-            ak._v2.contents.RecordArray(
-                fields=[column],
-                contents=[layout],
-            ),
-            highlevel=True,
-        )
+def from_rdataframe(data_frame, columns):
+    def supported(form):
+        if form.purelist_depth == 1:
+            # special case for a list of strings form
+            return isinstance(
+                form, (ak._v2.forms.ListOffsetForm, ak._v2.forms.NumpyForm)
+            )
+        else:
+            return isinstance(form, ak._v2.forms.ListOffsetForm) and supported(
+                form.content
+            )
```

Comment:
Independently checking

```
supported(T) = (is_list(T) or is_primitive(T)) or (is_list(T) and supported(T.content))
```

is completely equivalent to

```
supported(T) = is_primitive(T) or (is_list(T) and supported(T.content))
```

(including lists that are strings, since they're also list-of-primitive).

However, you're excluding fixed-length lists/RegularForms (why?). If there's a good reason for that, you also need to exclude fixed-length lists as NumpyForms with an inner shape.

```suggestion
    def supported(form):
        if isinstance(form, ak._v2.forms.NumpyForm) and form.inner_shape == ():
            return True
        else:
            return isinstance(form, ak._v2.forms.ListOffsetForm) and supported(
                form.content
            )
```

But coming back to the intention: are these the Forms you want to select? Often, if you're taking ListOffsetForm, then you also want ListForm and maybe even RegularForm (in which case, those NumpyForms with non-trivial inner shapes would be acceptable).

Is there some reason you can expect the desired list types to always be ListOffsetForm and not ListForm?

Reply:
The `Form` that is checked here is actually generated from the C++ type of the RDF columns such as nested `std::vectors` or nested `RVecs`, so it's chosen to be only `ListOffsetForm`.

Reply:
That would be a sufficient reason, and it also answers my question about other types (BitMaskedForm, etc.).